prawcore/sessions.py

```diff
@@ -152,19 +190,22 @@ def _make_request(self, data, files, json, method, params, retries, url):
             )
             return response, None
         except RequestException as exception:
-            if retries <= 1 or not isinstance(
+            if not retry_strategy_state.has_remaining_retries() or not isinstance(
                 exception.original_exception, self.RETRY_EXCEPTIONS
             ):
                 raise
             return None, exception.original_exception
 
     def _request_with_retries(
-        self, data, files, json, method, params, url, retries=3
+        self, data, files, json, method, params, url, retry_strategy_state=None
```

Comment:
Since instances of the class are immutable, it should be okay to be used as the default argument.

Reply:
The issue is that you can't use `self` because it's only bound in the body of the function.

Reply:
However, it might be more straightforward to have the initial state be an attribute of the object, rather than having a class and then instantiating later on.

Reply:
Ah right. I see. Let's do this, as is for now.